.github/templates/linux_ci_workflow.yml.j2

```diff
@@ -189,13 +189,39 @@ jobs:
           # Prune all of the docker images
           docker system prune -af
 
+  generate-test-matrix:
+    if: ${{ github.repository_owner == 'pytorch' }}
+    runs-on: ubuntu-18.04
+    env:
+      NUM_TEST_SHARDS: !{{ num_test_shards }}
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    container:
+      image: python:3.9
+    steps:
+      - name: Clone pytorch/pytorch
+        uses: actions/checkout@v2
+      - name: Generating test matrix
+        id: set-matrix
+        run: |
+          # outputting for debugging purposes
+          MATRIX=$(python .github/scripts/generate_pytorch_test_matrix.py)
+          echo "${MATRIX}"
+          echo "::set-output name=matrix::${MATRIX}"
```

Comment:
could you clarify the reason this has to be its own separate job?

Reply:
I copied this logic from the binary workflow logic--the idea is to have a matrix generator that could be used as the strategy of the test job. I wasn't sure there were other alternatives that'd let me pass the results of this as a matrix to another job.

Reply:
Eh I think keeping it as its own job is fine, makes it easier to see where dependencies like MATRIX variable are coming from